internal/configs/parser_config_dir.go

```diff
@@ -170,117 +180,40 @@ func (p *Parser) loadFiles(paths []string, override bool) ([]*File, hcl.Diagnost
+func (p *Parser) loadTestFiles(basePath string, paths []string) (map[string]*TestFile, hcl.Diagnostics) {
+	var diags hcl.Diagnostics
+
+	tfs := make(map[string]*TestFile)
+	for _, path := range paths {
+		tf, fDiags := p.LoadTestFile(path)
+		diags = append(diags, fDiags...)
+		if tf != nil {
+			// We index test files relative to the module they are testing, so
+			// the key is the relative path between basePath and path.
+			relPath, err := filepath.Rel(basePath, path)
+			if err != nil {
 				diags = append(diags, &hcl.Diagnostic{
-					Severity: hcl.DiagError,
-					Summary:  "Failed to read test directory",
-					Detail:   fmt.Sprintf("Test directory %s could not be read: %v.", testPath, err),
+					Severity: hcl.DiagWarning,
+					Summary:  "Failed to calculate relative path",
+					Detail:   fmt.Sprintf("Terraform could not calculate the relative path for test file %s and it has been skipped: %s", path, err),
 				})
-
-				// We'll also stop loading the rest of the config for this.
-				return
-			}
-
-		} else {
-			for _, testInfo := range infos {
-				if testInfo.IsDir() || IsIgnoredFile(testInfo.Name()) {
-					continue
-				}
-
-				if strings.HasSuffix(testInfo.Name(), ".tftest.hcl") || strings.HasSuffix(testInfo.Name(), ".tftest.json") {
-					tests = append(tests, filepath.Join(testPath, testInfo.Name()))
-				}
-			}
-		}
-
-	}
-
-	infos, err := p.fs.ReadDir(dir)
-	if err != nil {
-		diags = append(diags, &hcl.Diagnostic{
-			Severity: hcl.DiagError,
-			Summary:  "Failed to read module directory",
-			Detail:   fmt.Sprintf("Module directory %s does not exist or cannot be read.", dir),
-		})
-		return
-	}
-
-	for _, info := range infos {
-		if info.IsDir() {
-			// We only care about terraform configuration files.
-			continue
-		}
-
-		name := info.Name()
-		ext := fileExt(name)
-		if ext == "" || IsIgnoredFile(name) {
-			continue
-		}
-
-		if ext == ".tftest.hcl" || ext == ".tftest.json" {
-			if includeTests {
-				tests = append(tests, filepath.Join(dir, name))
+				continue
 			}
-			continue
-		}
-
-		baseName := name[:len(name)-len(ext)] // strip extension
-		isOverride := baseName == "override" || strings.HasSuffix(baseName, "_override")
-
-		fullPath := filepath.Join(dir, name)
-		if isOverride {
-			override = append(override, fullPath)
-		} else {
-			primary = append(primary, fullPath)
+			tfs[relPath] = tf
 		}
 	}
 
-	return
+	return tfs, diags
 }
 
-func (p *Parser) loadTestFiles(basePath string, paths []string) (map[string]*TestFile, hcl.Diagnostics) {
+func (p *Parser) loadQueryFiles(basePath string, paths []string) (map[string]*QueryFile, hcl.Diagnostics) {
+	files := make(map[string]*QueryFile)
 	var diags hcl.Diagnostics
 
-	tfs := make(map[string]*TestFile)
 	for _, path := range paths {
-		tf, fDiags := p.LoadTestFile(path)
+		f, fDiags := p.LoadQueryFile(path)
 		diags = append(diags, fDiags...)
-		if tf != nil {
+		if f != nil {
 			// We index test files relative to the module they are testing, so
```

Comment:
This comment needs updating, but query files are not external to the module like tests, so is there anything else to change?

Reply:
Thanks!. copy-paste error. query files need no relative indexing, because at the moment, we expect them to be at the root of the configuration.